drivers/ethernet/eth_lan9250.c

```diff
@@ -0,0 +1,713 @@
+				LOG_DBG("LINK DOWN");
+				net_eth_carrier_off(context->iface);
+			}
+		}
+
+		if ((int_sts & LAN9250_INT_STS_RSFL) != 0) {
+			lan9250_write_sys_reg(context->dev, LAN9250_INT_STS, LAN9250_INT_STS_RSFL);
+			lan9250_rx(context->dev);
+		}
+
+		/* Re-enable interrupts */
+		lan9250_write_sys_reg(context->dev, LAN9250_INT_EN, ier);
+	}
+}
+
+static enum ethernet_hw_caps lan9250_get_capabilities(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return ETHERNET_LINK_10BASE_T | ETHERNET_LINK_100BASE_T;
+}
+
+static void lan9250_iface_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct lan9250_runtime *context = dev->data;
+
+	net_if_set_link_addr(iface, context->mac_address, sizeof(context->mac_address),
+			     NET_LINK_ETHERNET);
+	context->iface = iface;
+	ethernet_init(iface);
+
+	net_if_carrier_off(iface);
+}
+
+static const struct ethernet_api api_funcs = {
+	.iface_api.init = lan9250_iface_init,
+	.get_capabilities = lan9250_get_capabilities,
+	.send = lan9250_tx,
+};
+
+static int lan9250_init(const struct device *dev)
+{
+	const struct lan9250_config *config = dev->config;
+	struct lan9250_runtime *context = dev->data;
+
+	context->dev = dev;
+
+	/* SPI config */
+	if (!spi_is_ready_dt(&config->spi)) {
+		LOG_ERR("SPI master port %s not ready", config->spi.bus->name);
+		return -EINVAL;
+	}
+
+	/* Initialize GPIO */
+	if (!gpio_is_ready_dt(&config->interrupt)) {
+		LOG_ERR("GPIO port %s not ready", config->interrupt.port->name);
+		return -EINVAL;
+	}
+
+	if (gpio_pin_configure_dt(&config->interrupt, GPIO_INPUT)) {
+		LOG_ERR("Unable to configure GPIO pin %u", config->interrupt.pin);
+		return -EINVAL;
+	}
+
+	gpio_init_callback(&(context->gpio_cb), lan9250_gpio_callback, BIT(config->interrupt.pin));
+	if (gpio_add_callback(config->interrupt.port, &(context->gpio_cb))) {
+		return -EINVAL;
+	}
+
+	gpio_pin_interrupt_configure_dt(&config->interrupt, GPIO_INT_EDGE_TO_ACTIVE);
+	lan9250_spi_check(dev);
+	lan9250_sw_reset(dev);
+	lan9250_configure(dev);
+	lan9250_set_macaddr(dev);
+
+	k_thread_create(&context->thread, context->thread_stack,
+			CONFIG_ETH_LAN9250_RX_THREAD_STACK_SIZE, lan9250_thread, context, NULL,
+			NULL, K_PRIO_COOP(CONFIG_ETH_LAN9250_RX_THREAD_PRIO), 0, K_NO_WAIT);
+
+	LOG_INF("LAN9250 Initialized");
+
+	return 0;
+}
+
+static struct lan9250_runtime lan9250_0_runtime = {
+	.mac_address = DT_INST_PROP(0, local_mac_address),
+	.tx_rx_sem = Z_SEM_INITIALIZER(lan9250_0_runtime.tx_rx_sem, 1, UINT_MAX),
+	.int_sem = Z_SEM_INITIALIZER(lan9250_0_runtime.int_sem, 0, UINT_MAX),
+};
+
+static const struct lan9250_config lan9250_0_config = {
+	.spi = SPI_DT_SPEC_INST_GET(0, SPI_WORD_SET(8), 0),
+	.interrupt = GPIO_DT_SPEC_INST_GET(0, int_gpios),
+	.timeout = CONFIG_ETH_LAN9250_BUF_ALLOC_TIMEOUT,
+};
+
+ETH_NET_DEVICE_DT_INST_DEFINE(0, lan9250_init, NULL, &lan9250_0_runtime, &lan9250_0_config,
+			      CONFIG_ETH_INIT_PRIORITY, &api_funcs, NET_ETH_MTU);
```

Comment:
Why limit to a single instance? Wrap a define macro like the most ethernet drivers and use `DT_INST_FOREACH_STATUS_OKAY`

Reply:
done :)